examples/remote-sensing/helpers/processing_helpers.py

```diff
@@ -0,0 +1,203 @@
+"""
+=================================
+Processing Remote Sensing Helpers
+=================================
+
+This file contains helper functions for handling remote sensing processes
+"""
+from math import ceil
+
+import numpy as np
+import numpy.linalg as la
+from numpy.lib.stride_tricks import sliding_window_view
+from numpy.typing import ArrayLike
+from sklearn.base import BaseEstimator, TransformerMixin
+
+
+###############################################################################
+
+def pca_image(image: ArrayLike, n_components: int):
+    """ A function that centers data and applies PCA on an image.
+
+    Parameters
+    ----------
+    image : ndarray, shape (n_rows, n_columns, n_features)
+        An image.
+    n_components : int
+        Number of components to keep.
+
+    Written by Antoine Collas for:
+    https://github.com/antoinecollas/pyCovariance/
+    """
+    # center pixels
+    h, w, p = image.shape
+    X = image.reshape((h*w, p))
+    mean = np.mean(X, axis=0)
+    image = image - mean
+    X = X - mean
+    # check pixels are centered
+    assert (np.abs(np.mean(X, axis=0)) < 1e-8).all()
+
+    # apply PCA
+    SCM = (1/len(X))*X.conj().T@X
+    d, Q = la.eigh(SCM)
+    reverse_idx = np.arange(len(d)-1, -1, step=-1)
+    Q = Q[:, reverse_idx]
+    Q = Q[:, :n_components]
+    image = image@Q
```

Comment:
I believe @antoinecollas used this because it's faster to apply over the multidimensional image.

In my case, after the sliding windows, the problem is that there are many redundant data because of the overlap. And even if there weren't the way the data is shaped is not exactly `(n_features, n_samples)'. One could always do something like vectorising image at first then pcq then shaping it back to an image with less features then sliding windows (it would be tricky to a sliding windows over vectorised version where we converted indexes of rows, columns to a 1D index) but since Antoine already coded the PCA over images I simply used that.

Reply:
I think it's better to use sklearn. Indeed, it means doing reshape -> PCA -> reshape.

Reply:
I am sorry, I just remembered that scikit-learn doesn't support complex data.
So, it is better to keep this implementationif you want to process complex images.

Reply:
Ok So I propose to use `sklearn.decomposition.PCA` if data is real and `pca_image` if complex. Would that work?

Reply:
Do you want to deal with complex-valued data? In this case, you'll also have problems with every clustering/classifier from sklearn. Does Pyriemann handle complex-valued data well?

Reply:
Yes, pyRiemann can process complex-valued data, see #204

For the PCA code, it's up to you whether you want to process complex radar images.

> So I propose to use `sklearn.decomposition.PCA` if data is real and `pca_image` if complex. Would that work?

Yes, you can do that.